client/lib/signup/step-actions.js

```diff
@@ -53,74 +137,31 @@ function setThemeOnSite( callback, { siteSlug }, { themeSlug } ) {
 }
 
 module.exports = {
-	addDomainItemsToCart( callback, dependencies, { domainItem, googleAppsCartItem, isPurchasingItem, siteUrl, themeSlug, themeItem } ) {
-		wpcom.undocumented().sitesNew( {
-			blog_name: siteUrl,
-			blog_title: siteUrl,
-			options: {
-				theme: dependencies.theme
-			},
-			validate: false,
-			find_available_url: isPurchasingItem
-		}, function( error, response ) {
-			if ( error ) {
-				return callback( error );
-			}
+	addDomainItemsToCart: addDomainItemsToCart,
 
-			const siteSlug = response.blog_details.blogname + '.wordpress.com';
-			const isFreeThemePreselected = themeSlug && ! themeItem;
-			const providedDependencies = {
-				siteSlug,
-				domainItem,
-				themeItem,
-			};
-			const addToCartAndProceed = () => {
-				let newCartItems = [];
-
-				if ( domainItem ) {
-					newCartItems = [ ...newCartItems, domainItem ];
-				}
-				if ( googleAppsCartItem ) {
-					newCartItems = [ ...newCartItems, googleAppsCartItem ];
-				}
-				if ( themeItem ) {
-					newCartItems = [ ...newCartItems, themeItem ];
-				}
-
-				if ( newCartItems.length ) {
-					SignupCart.addToCart( siteSlug, newCartItems, function( cartError ) {
-						callback( cartError, providedDependencies );
-					} );
-				} else {
-					callback( [], providedDependencies );
-				}
-			};
-
-			if ( ! user.get() && isFreeThemePreselected ) {
-				return setThemeOnSite( addToCartAndProceed, { siteSlug }, { themeSlug } );
-			}
-
-			if ( user.get() && isFreeThemePreselected ) {
-				return fetchSitesAndUser( siteSlug, setThemeOnSite.bind( this, addToCartAndProceed, { siteSlug }, { themeSlug } ) );
-			} else if ( user.get() ) {
-				return fetchSitesAndUser( siteSlug, addToCartAndProceed );
+	addDomainAndPlanFreeTrialToCart( callback, dependencies, data ) {
```

Comment:
Now that we start a free trial during this action (instead of adding it to the cart), we should probably rename this.